pkgs/servers/nextcloud/packages/generate.sh

```diff
@@ -8,7 +8,55 @@ set -x
 
 export NEXTCLOUD_VERSIONS=$(nix-instantiate --eval -E 'import ./nc-versions.nix {}' -A e)
 
-APPS=`cat nextcloud-apps.json | jq -r 'keys|.[]' | sed -z 's/\n/,/g;s/,$/\n/'`
+APPS=$(jq -r 'keys|.[]' nextcloud-apps.json | sed -z 's/\n/,/g;s/,$/\n/')
 
-nc4nix -apps $APPS
-rm *.log
+for v in ${NEXTCLOUD_VERSIONS//,/ }; do
+    # Get major version and back up previous major version apps file
+    v=$(sed -e 's/^"//' -e 's/"$//' <<<"$v")
+    MAJOR=${v%.*.*}
+    MAJOR_FILE="$MAJOR".json
+    mv "$MAJOR_FILE" "$MAJOR_FILE".bak
+
+    # Download current apps file from Nextcloud's official servers
+    APPS_PER_VERSION=${v}.json
+    curl "https://apps.nextcloud.com/api/v1/platform/${v}/apps.json" -o "$APPS_PER_VERSION"
+
+    # Add a starting bracket to the apps file for this version
+    echo '{' >"$MAJOR_FILE".tmp
+    for a in ${APPS//,/ }; do
+        echo "Fetching $a"
+        # Ensure the app exists in the file
+        if [ "$(jq -r ".[] | select(.id == \"${a}\")" "$APPS_PER_VERSION")" != "" ]; then
+            # Get all of our variables
+            VERSION=$(jq -r ".[] | select(.id == \"${a}\") | .releases[0].version" "$APPS_PER_VERSION")
```

Comment:
This just picks the first item in releases, which might not actually be the most recent version. See e.g. the calendar app.